src/SonarScanner.MSBuild/BootstrapperClass.cs

```diff
@@ -129,17 +174,33 @@ private int PostProcess()
         /// <summary>
         /// Copies DLLs needed by the targets file that is loaded by MSBuild to the project's .sonarqube directory
         /// </summary>
-        private void CopyDLLs()
+        private bool CopyDlls()
         {
-            var binDirPath = Path.Combine(bootstrapSettings.TempDirectory, "bin");
+            var binDirPath = Path.Combine(this.bootstrapSettings.TempDirectory, "bin");
             Directory.CreateDirectory(binDirPath);
             string[] dllsToCopy = { "SonarScanner.MSBuild.Common.dll", "SonarScanner.MSBuild.Tasks.dll" };
 
             foreach (var dll in dllsToCopy)
             {
-                var dllPath = Path.Combine(bootstrapSettings.ScannerBinaryDirPath, dll);
-                File.Copy(dllPath, Path.Combine(binDirPath, dll));
+                var from = Path.Combine(this.bootstrapSettings.ScannerBinaryDirPath, dll);
+                var to = Path.Combine(binDirPath, dll);
+
+                if (!File.Exists(to))
+                {
+                    File.Copy(from, to);
+                }
+                else if (this.getAssemblyVersion(from).CompareTo(this.getAssemblyVersion(to)) != 0)
```

Comment:
the camelCase methods look awkward... Perhaps we should change the naming convention for private function fields to PascalCase

Reply:
Could `getAssemblyVersion` return null or throw exception? Perhaps we should handle these cases.